src/internal/connector/data_collections_test.go

```diff
@@ -302,6 +303,77 @@ func (suite *ConnectorCreateExchangeCollectionIntegrationSuite) TestMailFetch()
 	}
 }
 
+func (suite *ConnectorCreateExchangeCollectionIntegrationSuite) TestDelta() {
+	ctx, flush := tester.NewContext()
+	defer flush()
+
+	var (
+		userID = tester.M365UserID(suite.T())
+		gc     = loadConnector(ctx, suite.T(), Users)
+	)
+
+	tests := []struct {
+		name  string
+		scope selectors.ExchangeScope
+	}{
+		{
+			name: "Mail",
+			scope: selectors.NewExchangeBackup().MailFolders(
+				[]string{userID},
+				[]string{exchange.DefaultMailFolder},
+				selectors.PrefixMatch(),
+			)[0],
+		},
+		{
+			name: "Contacts",
+			scope: selectors.NewExchangeBackup().ContactFolders(
+				[]string{userID},
+				[]string{exchange.DefaultContactFolder},
+				selectors.PrefixMatch(),
+			)[0],
+		},
+	}
+	for _, test := range tests {
+		suite.T().Run(test.name, func(t *testing.T) {
+			// get collections without providing any delta history (ie: full backup)
+			collections, err := gc.createExchangeCollections(ctx, test.scope, nil)
+			require.NoError(t, err)
+			assert.Less(t, 1, len(collections), "retrieved metadata and data collections")
+
+			var metadata data.Collection
+
+			for _, coll := range collections {
+				if coll.FullPath().Service() == path.ExchangeMetadataService {
+					metadata = coll
+				}
+			}
+
+			require.NotNil(t, metadata, "collections contains a metadata collection")
+
+			_, deltas, err := exchange.ParseMetadataCollections(ctx, []data.Collection{metadata})
```

Comment:
this is kind of sixes, but you could just call `gc.DataCollections()` 2x instead of manually parsing the metadata collection and calling `gc.createExchangeCollections()`. Whatever you prefer works

Reply:
Gonna keep it as is for now. We can look into refinements like this on the upcoming refactor.